packages/@mantine/hooks/src/use-resize-observer/use-resize-observer.ts

```diff
@@ -30,7 +30,31 @@ export function useResizeObserver<T extends HTMLElement = any>(options?: ResizeO
 
               frameID.current = requestAnimationFrame(() => {
                 if (ref.current) {
-                  setRect(entry.contentRect);
+                  const boxSize = entry.borderBoxSize?.[0] || entry.contentBoxSize?.[0];
+                  if (boxSize) {
+                    const style = window.getComputedStyle(ref.current);
```

Comment:
window.getComputedStyle is an expensive operation, considering that it is called on every resize, it is not a good solution. I'd rather not handle writing mode at all (vertical writing mode is very rare and Mantine does not have a single user with it).